src/components/ChangePlan/PlanCalculator.test.js

```diff
@@ -0,0 +1,374 @@
+        <IntlProvider onError={() => null}>
+          <BrowserRouter>
+            <PlanCalculator />
+          </BrowserRouter>
+        </IntlProvider>
+      </AppServicesProvider>,
+    );
+
+    // Assert
+    await waitFor(() => {
+      expect(container.querySelector('.dp-wrap-subscription')).not.toBeInTheDocument();
+    });
+  });
+
+  it('should show message when it has a discount applied', async () => {
+    // Arrange
+    const planServiceDouble = {
+      getPlanList: () => [],
+      mapCurrentPlanFromTypeOrId: () => {},
+      getPlanTypes: () => ['prepaid', 'subscribers', 'monthly-deliveries'],
+      getBuyUrl: () => '',
+      getPlans: () => [
+        {
+          type: 'prepaid',
+          id: 1,
+          name: '1500-CREDITS',
+          credits: 1500,
+          price: 15,
+          featureSet: 'standard',
+          billingCycleDetails: [
+            {
+              id: 56,
+              idPlan: 11,
+              paymentType: 'CC',
+              discountPercentage: 5,
+              billingCycle: 'quarterly',
+            },
+          ],
+        },
+      ],
+    };
+
+    // Act
+    const { getByText } = render(
+      <AppServicesProvider forcedServices={{ ...dependencies, planService: planServiceDouble }}>
+        <IntlProvider onError={() => null}>
+          <BrowserRouter>
+            <PlanCalculator />
+          </BrowserRouter>
+        </IntlProvider>
+      </AppServicesProvider>,
+    );
+
+    // Assert
+    await waitFor(() => {
+      expect(getByText('plan_calculator.with_quarterly_discount')).toBeInTheDocument();
+    });
+  });
+
+  it('should not show message when it has not a discount applied', async () => {
+    // Arrange
+    const planServiceDouble = {
+      getPlanList: () => [],
+      mapCurrentPlanFromTypeOrId: () => {},
+      getPlanTypes: () => ['prepaid', 'subscribers', 'monthly-deliveries'],
+      getBuyUrl: () => '',
+      getPlans: () => [
+        {
+          type: 'prepaid',
+          id: 1,
+          name: '1500-CREDITS',
+          credits: 1500,
+          price: 15,
+          featureSet: 'standard',
+        },
+      ],
+    };
+
+    // Act
+    const { queryByText } = render(
+      <AppServicesProvider forcedServices={{ ...dependencies, planService: planServiceDouble }}>
+        <IntlProvider onError={() => null}>
+          <BrowserRouter>
+            <PlanCalculator />
+          </BrowserRouter>
+        </IntlProvider>
+      </AppServicesProvider>,
+    );
+
+    // Assert
+    await waitFor(() => {
+      expect(queryByText('plan_calculator.with_quarterly_discount')).toBe(null);
+      expect(queryByText('plan_calculator.with_half_yearly_discount')).toBe(null);
+      expect(queryByText('plan_calculator.with_yearly_discount')).toBe(null);
+    });
+  });
+
+  it('should calc price with discount', async () => {
+    // Arrange
+    const planServiceDouble = {
```

Comment:
Perhaps instead of redefining this in each test, you could have a function that returns a service with, all plan types available or only prepaid, and so on. To improve readability. What do you think?

Reply:
Yes, or maybe I can define only getPlans method in each test and have just once the other ones (because are always the same)... I mean.. to be clear what is the context in each test.

Reply:
```
    const planServiceDouble = {
      ...planServiceDoubleBase,
      getPlans: () => [
        {
          type: 'prepaid',
          id: 1,
          name: '1500-CREDITS',
          credits: 1500,
          price: 15,
          featureSet: 'standard',
        },
      ],
    };
```